src/MSBuild/XMake.cs

```diff
@@ -1427,17 +1532,29 @@ private static void ResetGatheringSwitchesState()
                             {
                                 if (graphBuildOptions != null)
                                 {
-                                    (result, exception) = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    graphResult = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    ProjectGraphEntryPoint entryPoint = graphBuildRequest.ProjectGraphEntryPoints.First();
+                                    if (!entryPoint.GlobalProperties.ContainsKey(PropertyNames.IsGraphBuild))
+                                    {
+                                        entryPoint.GlobalProperties[PropertyNames.IsGraphBuild] = "true";
+                                    }
+
+                                    result = graphResult.ResultsByNode.First(
```

Comment:
Likewise if there is more than one match here things are broken, right?
```suggestion
                                    result = graphResult.ResultsByNode.Single(
```